src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -982,17 +982,59 @@ class ArrayGenerator : public Generator<bsoncxx::array::value> {
     const UniqueGenerator<int64_t> _nTimesGen;
 };
 
+class ConcatGenerator : public Generator<bsoncxx::array::value> {
+
+public:
+    ConcatGenerator(const Node& node,
+                    GeneratorArgs generatorArgs,
+                    std::map<std::string, Parser<UniqueAppendable>> parsers)
+        : _rng{generatorArgs.rng}, _id{generatorArgs.actorId} {
+        if (!node["arrays"].isSequence()) {
+            std::stringstream msg;
+            msg << "Malformed node for join array. Not a sequence " << node;
+            BOOST_THROW_EXCEPTION(InvalidValueGeneratorSyntax(msg.str()));
+        }
+        for (const auto&& [k, v] : node["arrays"]) {
```

Comment:
I think you should create a function UniqueGenerator<bsoncxx::array::value> BsonArrayGenerator similar to IntGenerator https://github.com/mongodb/genny/blob/89ce68b88ad27c35f80f98cf11078564e20c74c1/src/value_generators/src/DocumentGenerator.cpp#L1373, doubleGenerator, etc. 

Then you can pass each element in node["arrays"] to that function. That will enable nested concat generators, and also support the two generators here. And if anyone adds a new array generator you will automatically be able to use it within a concat generator.